src/compiler/checker.ts

```diff
@@ -32180,9 +32184,9 @@ namespace ts {
         function assignParameterType(parameter: Symbol, type?: Type) {
             const links = getSymbolLinks(parameter);
             if (!links.type) {
-                const declaration = parameter.valueDeclaration as ParameterDeclaration;
-                links.type = type || getWidenedTypeForVariableLikeDeclaration(declaration, /*reportErrors*/ true);
-                if (declaration.name.kind !== SyntaxKind.Identifier) {
+                const declaration = parameter.valueDeclaration as ParameterDeclaration | undefined;
+                links.type = type || (declaration ? getWidenedTypeForVariableLikeDeclaration(declaration, /*reportErrors*/ true) : getTypeOfSymbol(parameter));
+                if (declaration && declaration.name.kind !== SyntaxKind.Identifier) {

```

Comment:
does `declaration?.name.kind` work here?

Reply:
Nope - the subsequent `declaration.name` accesses aren't correctly narrowed if I do that, apparently.

Reply:
You'd have to invert the condition, otherwise you'll get `true` when `declaration` is present but not an identifier

Reply:
(Because `undefined` is not equal to `SyntaxKind.Identifier`)